### PR TITLE
Update collector to use common data types from telemetry-exporter

### PR DESCRIPTION
**Pull request by shaun-nx**
### Proposed changes

This PR integrates the common Data structure defined in https://github.com/nginxinc/telemetry-exporter/pull/27 to allow NIC to generate its Schema.

This PR also adds two `go generate` annotations to the `Data` and `NICResourceCounts` structs to generate the required `Attributes` function. This ensures the data is of type [Exportable](https://github.com/nginxinc/telemetry-exporter/blob/main/pkg/telemetry/exporter.go#L27-L30)

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -139,3 +141,5 @@ require (
 )
 
 replace github.com/golang/glog => github.com/nginxinc/glog v1.1.2
+
+replace github.com/nginxinc/telemetry-exporter => /Users/s.odonovan/github.com/nginxinc/telemetry-exporter
```
Comment (shaun-nx) [inline go.mod:145]: Will remove this line as soon as https://github.com/nginxinc/telemetry-exporter/pull/27 is merged

File: internal/telemetry/collector.go [hunk 2]
```diff
@@ -113,16 +113,21 @@ func (c *Collector) BuildReport(ctx context.Context) (Data, error) {
...
+
+	if d.ClusterVersion, err = c.ClusterVersion(); err != nil {
 		glog.Errorf("Error collecting telemetry data: K8s Version: %v", err)
 	}
-	return d, err
+
+	// TODO: Get Cluster (k8s) platform. e.g. EKS, AWS, Openshift, etc...
+
+	// TODO: Get InstallationID
+	// example of how NGF gets this ID https://github.com/nginxinc/nginx-gateway-fabric/blob/f33db51fc9e05ccf98fc8cdae100772a5cc6775e/internal/mode/static/telemetry/collector.go#L244-L248
```
Comment (shaun-nx) [inline internal/telemetry/collector.go:131]: Both Deployment Platform and InstallationID are the last two common data points that NIC needs to gather.

Both are addresses in https://github.com/nginxinc/kubernetes-ingress/issues/5144 and https://github.com/nginxinc/kubernetes-ingress/issues/4882 respectfully

Comment (shaun-nx) [reply internal/telemetry/collector.go:131]: https://github.com/nginxinc/kubernetes-ingress/issues/4882 still needs discussion as how NIC and NGF approach these may be different.

See https://github.com/nginxinc/nginx-gateway-fabric/blob/f33db51fc9e05ccf98fc8cdae100772a5cc6775e/internal/mode/static/telemetry/collector.go#L244-L248 for reference on NGF's approach

File: go.mod (hunk 1 above)
Comment (danielnginx) [reply go.mod:144]: https://github.com/nginxinc/telemetry-exporter/pull/27 is merged

**Review by jjngx — APPROVED** [review]

**Review by vepatel — APPROVED** [review]

**Pull request merged**